odra-schema/src/lib.rs

```diff
@@ -219,3 +255,175 @@ fn call_method(
+        match custom_struct {
+            casper_contract_schema::CustomType::Struct { name, members, .. } => {
+                assert_eq!(name, "struct1".into());
+                assert_eq!(members.len(), 1);
+            }
+            _ => panic!("Expected CustomType::Struct")
+        }
+    }
+
+    #[test]
+    fn test_custom_enum() {
+        let variant = super::enum_variant("variant1", 1);
+        let custom_enum = super::custom_enum("enum1", vec![variant]);
+        match custom_enum {
+            casper_contract_schema::CustomType::Enum { name, variants, .. } => {
+                assert_eq!(name, "enum1".into());
+                assert_eq!(variants.len(), 1);
+            }
+            _ => panic!("Expected CustomType::Enum")
+        }
+    }
+
+    #[test]
+    fn test_event() {
+        let event = super::event("event1");
+        assert_eq!(event.name, "event1");
+    }
+
+    #[test]
+    fn test_error() {
+        let error = super::error("error1", "description", 1);
+        assert_eq!(error.name, "error1");
+        assert_eq!(error.description, Some("description".to_string()));
+        assert_eq!(error.discriminant, 1);
+    }
+
+    mockall::mock! {
+        TestSchema {}
+        impl SchemaEntrypoints for TestSchema {
+            fn schema_entrypoints() -> Vec<Entrypoint>;
+        }
+
+        impl SchemaEvents for TestSchema {
+            fn schema_events() -> Vec<Event>;
+            fn custom_types() -> Vec<Option<CustomType>>;
+        }
+
+        impl SchemaCustomTypes for TestSchema {
+            fn schema_types() -> Vec<Option<CustomType>>;
+        }
+
+        impl SchemaErrors for TestSchema {
+            fn schema_errors() -> Vec<UserError>;
+        }
+    }
+
+    #[test]
+    fn test_schema() {
+        let entry_points_ctx = MockTestSchema::schema_entrypoints_context();
+        entry_points_ctx.expect().returning(|| {
+            vec![super::entry_point::<u32>(
+                "entry1",
+                "description",
+                true,
+                vec![super::argument::<u32>("arg1")]
+            )]
+        });
+
+        let events_ctx = MockTestSchema::schema_events_context();
+        events_ctx
+            .expect()
+            .returning(|| vec![super::event("event1")]);
+
+        let custom_types_ctx = MockTestSchema::schema_types_context();
+        custom_types_ctx.expect().returning(|| {
+            vec![
+                Some(super::custom_struct(
+                    "struct1",
+                    vec![super::struct_member::<u32>("member1")]
+                )),
+                Some(super::custom_enum(
+                    "struct1",
+                    vec![super::enum_variant("variant1", 1)]
+                )),
+            ]
+        });
+
+        let errors_ctx = MockTestSchema::schema_errors_context();
+        errors_ctx.expect().returning(Vec::new);
+
+        let schema = super::schema::<MockTestSchema>(
```

Comment:
This could be easily done without mocks. Simply define `struct TestSchema` and implement  traits for it.